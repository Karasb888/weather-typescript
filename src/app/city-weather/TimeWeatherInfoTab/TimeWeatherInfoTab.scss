.weatherOption {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.icon {
    padding-right: 5px;
}

.optionName {
    font-size: 15.5px;
    font-family: 'Cooper-Hewitt';
    font-weight: 300;
}

.optionValue {
    font-size: 15px;
    font-family: 'Cooper-Hewitt';
    font-weight: 400;
    justify-self: flex-end;
    margin-left: auto;
}