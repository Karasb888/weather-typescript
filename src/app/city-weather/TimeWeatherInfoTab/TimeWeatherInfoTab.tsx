import * as React from 'react';
import { WeatherListObj } from '../../../store/weather';
import TempIcon from '../../icons/TempIcon';
import TempFeelsIcon from '../../icons/TempFeelsIcon';
import CloudsIcon from '../../icons/CloudsIcon';
import WindIcon from '../../icons/WindIcon';
import * as styles from './TimeWeatherInfoTab.scss';

interface Props {
    timeWeather: WeatherListObj;
};

const TimeWeatherInfoTab: React.FC<Props> = ({ timeWeather }) => {
    const iconsProps = {
        height: '25',
        width: '25',
        svgClass: styles.icon
    };

    return (
        <div>
            <div className={styles.weatherOption}>
                <TempIcon
                    {...iconsProps}
                    viewBox="0 0 64 64"
                />
                <div className={styles.optionName}>Temperture:</div>
                <div className={styles.optionValue}>{timeWeather.main.temp}<span className={styles.celsius}>&#8451;</span></div>
            </div>
            <div className={styles.weatherOption}>
                <TempFeelsIcon
                    {...iconsProps}
                    viewBox="0 0 64 64"
                />
                <div className={styles.optionName}>Feels temperture:</div>
                <div className={styles.optionValue}>{timeWeather.main.feels_like}<span className={styles.celsius}>&#8451;</span></div>
            </div>
            <div className={styles.weatherOption}>
                <CloudsIcon
                    {...iconsProps}
                    viewBox="0 -87 463.83425 463"
                />
                <div className={styles.optionName}>Clouds:</div>
                <div className={styles.optionValue}>{timeWeather.clouds.all}%</div>
            </div>
            <div className={styles.weatherOption}>
                <WindIcon
                    {...iconsProps}
                    viewBox="0 0 512 512"
                />
                <div className={styles.optionName}>Wind:</div>
                <div className={styles.optionValue}>{timeWeather.wind.speed}m/s {timeWeather.wind.deg}deg</div>
            </div>
        </div>
    );
};

export default TimeWeatherInfoTab;
