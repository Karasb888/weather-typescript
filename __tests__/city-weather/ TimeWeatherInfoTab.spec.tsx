import * as React from 'react';
import { mockedStoreKyiv } from '../mockedStoreKyiv.data';
import { shallow } from 'enzyme';
import TimeWeatherInfoTab from '../../src/app/city-weather/TimeWeatherInfoTab/TimeWeatherInfoTab';

describe('++++ <TimeWeatherInfoTab /> ++++', () => {
    mockedStoreKyiv.weather.weatherCity[0].timeDayWeather.forEach((timeWeather) => {
        const props = { timeWeather };
        const wrapper = shallow(<TimeWeatherInfoTab {...props} />);

        it('Renders correctly with different props', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});