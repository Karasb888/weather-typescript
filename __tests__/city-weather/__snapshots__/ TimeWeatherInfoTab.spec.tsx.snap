// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <TimeWeatherInfoTab /> ++++ Renders correctly with different props 1`] = `
<div>
  <div
    className="weatherOption"
  >
    <TempIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Temperture:
    </div>
    <div
      className="optionValue"
    >
      8.8
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <TempFeelsIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Feels temperture:
    </div>
    <div
      className="optionValue"
    >
      4.86
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <CloudsIcon
      height="25"
      svgClass="icon"
      viewBox="0 -87 463.83425 463"
      width="25"
    />
    <div
      className="optionName"
    >
      Clouds:
    </div>
    <div
      className="optionValue"
    >
      53
      %
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <WindIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 512 512"
      width="25"
    />
    <div
      className="optionName"
    >
      Wind:
    </div>
    <div
      className="optionValue"
    >
      3
      m/s 
      217
      deg
    </div>
  </div>
</div>
`;

exports[`++++ <TimeWeatherInfoTab /> ++++ Renders correctly with different props 2`] = `
<div>
  <div
    className="weatherOption"
  >
    <TempIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Temperture:
    </div>
    <div
      className="optionValue"
    >
      7.24
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <TempFeelsIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Feels temperture:
    </div>
    <div
      className="optionValue"
    >
      2.49
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <CloudsIcon
      height="25"
      svgClass="icon"
      viewBox="0 -87 463.83425 463"
      width="25"
    />
    <div
      className="optionName"
    >
      Clouds:
    </div>
    <div
      className="optionValue"
    >
      99
      %
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <WindIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 512 512"
      width="25"
    />
    <div
      className="optionName"
    >
      Wind:
    </div>
    <div
      className="optionValue"
    >
      4.05
      m/s 
      224
      deg
    </div>
  </div>
</div>
`;

exports[`++++ <TimeWeatherInfoTab /> ++++ Renders correctly with different props 3`] = `
<div>
  <div
    className="weatherOption"
  >
    <TempIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Temperture:
    </div>
    <div
      className="optionValue"
    >
      5.26
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <TempFeelsIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Feels temperture:
    </div>
    <div
      className="optionValue"
    >
      0.2
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <CloudsIcon
      height="25"
      svgClass="icon"
      viewBox="0 -87 463.83425 463"
      width="25"
    />
    <div
      className="optionName"
    >
      Clouds:
    </div>
    <div
      className="optionValue"
    >
      97
      %
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <WindIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 512 512"
      width="25"
    />
    <div
      className="optionName"
    >
      Wind:
    </div>
    <div
      className="optionValue"
    >
      4.32
      m/s 
      201
      deg
    </div>
  </div>
</div>
`;

exports[`++++ <TimeWeatherInfoTab /> ++++ Renders correctly with different props 4`] = `
<div>
  <div
    className="weatherOption"
  >
    <TempIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Temperture:
    </div>
    <div
      className="optionValue"
    >
      4.22
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <TempFeelsIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 64 64"
      width="25"
    />
    <div
      className="optionName"
    >
      Feels temperture:
    </div>
    <div
      className="optionValue"
    >
      -1.76
      <span
        className="celsius"
      >
        ℃
      </span>
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <CloudsIcon
      height="25"
      svgClass="icon"
      viewBox="0 -87 463.83425 463"
      width="25"
    />
    <div
      className="optionName"
    >
      Clouds:
    </div>
    <div
      className="optionValue"
    >
      100
      %
    </div>
  </div>
  <div
    className="weatherOption"
  >
    <WindIcon
      height="25"
      svgClass="icon"
      viewBox="0 0 512 512"
      width="25"
    />
    <div
      className="optionName"
    >
      Wind:
    </div>
    <div
      className="optionValue"
    >
      5.48
      m/s 
      212
      deg
    </div>
  </div>
</div>
`;
